src/Api.elm

```diff
@@ -68,157 +73,106 @@ facebookAuthUrl =
+assembleMe document =
+    JsonApi.Documents.primaryResource document
+        :> \meResource ->
+            JsonApi.Resources.attributes decodeMeAttributes meResource
+                :> \name ->
+                    Ok { name = name }
+
+
+decodeMeAttributes : Decoder String
+decodeMeAttributes =
+    Decode.at [ "name" ] Decode.string
+
+
+assembleProposal : JsonApi.Document -> Result String Proposal
+assembleProposal document =
+    JsonApi.Documents.primaryResource document
+        :> \proposalResource ->
+            JsonApi.Resources.attributes decodeProposalAttributes proposalResource
+                :> \( title, body ) ->
+                    JsonApi.Resources.relatedResource "author" proposalResource
+                        :> \participantResource ->
+                            JsonApi.Resources.attributes decodeParticipantAttributes participantResource
+                                :> \name ->
+                                    Ok
+                                        { id = JsonApi.Resources.id proposalResource
+                                        , title = title
+                                        , body = body
+                                        , author =
+                                            { id = JsonApi.Resources.id participantResource
+                                            , name = name
+                                            }
+                                        }
+
+
+decodeProposalAttributes : Decoder ( String, String )
+decodeProposalAttributes =
+    Decode.object2 (,)
+        (Decode.at [ "title" ] Decode.string)
+        (Decode.at [ "body" ] Decode.string)
+
+
+decodeParticipantAttributes : Decoder String
+decodeParticipantAttributes =
+    Decode.at [ "name" ] Decode.string
 
 
 encodeProposalInput : ProposalInput -> String
 encodeProposalInput proposalInput =
-    Encode.object
-        [ ( "data"
-          , Encode.object
-                [ ( "type", Encode.string "proposal" )
-                , ( "attributes"
-                  , Encode.object
-                        [ ( "title", Encode.string proposalInput.title )
-                        , ( "body", Encode.string proposalInput.body )
-                        ]
-                  )
-                ]
-          )
-        ]
-        |> Encode.encode 0
+    JsonApi.Extra.encodeDocument "proposal" <|
+        Encode.object
+            [ ( "title", Encode.string proposalInput.title )
+            , ( "body", Encode.string proposalInput.body )
+            ]
 
 
 
 -- COMMANDS
 
 
-authenticateCmd : String -> (Msg -> a) -> Cmd a
-authenticateCmd authCode wrapMsg =
-    let
-        body =
-            "{\"auth_code\": \"" ++ authCode ++ "\"}"
-
-        requestTask =
-            exchangeAuthCodeForToken body
-    in
-        Task.perform AuthFailed GotAccessToken requestTask
-            |> Cmd.map wrapMsg
+authenticate : String -> (Msg -> a) -> Cmd a
+authenticate authCode wrapMsg =
```

Comment:
Just took a quick look at the effects libraries at the front page of http://package.elm-lang.org/

No, there is no such convention there.